[AbdulrahmanAlRammah] iP

AbdulrahmanAlRammah opened the pull request:
### Twilight 😁 

> "Every big change starts with a small step"

Twilight helps **_you_** keep track of:
- todo
- event
- deadlines

How to use:
1. launch program
2. enter commands as needed
3. say "bye" to close

Remaining improvements:
- [ ] Add help command
- [ ] Add GUI

For issues:[link](https://github.com/nus-cs2103-AY2425S1/forum)


Example of constructor for a command class:
```java
public AddCommand(String details) {
    this.details = details;
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

AbdulrahmanAlRammah committed:
added greetings and goodbye

AbdulrahmanAlRammah committed:
Added echo

AbdulrahmanAlRammah committed:
Added the list and add capabilities

AbdulrahmanAlRammah committed:
Level 3 with completion of tasks

AbdulrahmanAlRammah committed:
Added different task types

AbdulrahmanAlRammah committed:
Semiautomated Testing

AbdulrahmanAlRammah committed:
Fixed 2 common errors

AbdulrahmanAlRammah committed:
Added deletion

AbdulrahmanAlRammah committed:
Add more OOP

AbdulrahmanAlRammah committed:
Finalize Storage addition

AbdulrahmanAlRammah committed:
Merge branch 'branch-Level-7'

AbdulrahmanAlRammah committed:
Store dates in deadline

AbdulrahmanAlRammah committed:
Merge branch 'branch-Level-8'

AbdulrahmanAlRammah committed:
Add OOP and account for misinput

AbdulrahmanAlRammah committed:
Add twilight package

AbdulrahmanAlRammah committed:
Merge branch 'add-gradle-support'

AbdulrahmanAlRammah committed:
Add JUnit testing

AbdulrahmanAlRammah committed:
Create JAR file

AbdulrahmanAlRammah committed:
Add Javadoc comments

AbdulrahmanAlRammah committed:
Comply to coding standard

AbdulrahmanAlRammah committed:
Add query

AbdulrahmanAlRammah committed:
Merge branch 'branch-A-Javadoc'

AbdulrahmanAlRammah committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/twilight/Deadline.java
#	src/main/java/twilight/Task.java
#	src/main/java/twilight/Todo.java

AbdulrahmanAlRammah committed:
Fix Grammar in Deadline JavaDoc

AbdulrahmanAlRammah committed:
Merge branch 'branch-Level-9'

yizhong187 commented on src/main/java/twilight/AddCommand.java:
```diff
@@ -0,0 +1,36 @@
+/**
+ * Represents a Command to add a particular instance of an event.
+ */
+public class AddCommand extends Command {
```
Should `AddCommand` be renamed to `AddEventCommand` for **add**ed clarity?

AbdulrahmanAlRammah replied:
I humbly disagree. Since there is an event task this may end up adding confusion as the addcommand class can add any of todo, event, deadline.

yizhong187 replied:
Oh sorry i meant `AddTaskCommand` since `Task` is the parent class of all 3.

yizhong187 commented on src/main/java/twilight/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    @Override
+    public String toStorage() {
```
Maybe `toStorageString` will be a more apt name as it is returning a string? Similarly for the other Command subclasses.

AbdulrahmanAlRammah replied:
Agreed.

yizhong187 commented on src/main/java/twilight/Event.java:
```diff
@@ -0,0 +1,26 @@
+package twilight;
+
+public class Event extends Task {
+    protected String starttime;
```
Should `starttime` be in pascalCase like `startTime`?

yizhong187 commented on src/main/java/twilight/MarkingCommand.java:
```diff
@@ -0,0 +1,32 @@
+import java.io.IOException;
+
+public class MarkingCommand extends Command {
+    protected int type;
```
Since type only has the value of 1 and 2, will it be better to declare an enum for it?

AbdulrahmanAlRammah replied:
The only problem with this, is that I would have to define the enum in the task class and then check the type of input to convert to an enum and it feels like the enum adds extra steps instead of solving anything, but this is probably because I am unfamiliar with enums and can't see the benefit.

yizhong187 commented on src/main/java/twilight/Parser.java:
```diff
@@ -0,0 +1,53 @@
+            while (!COMMANDS[i].equals(command)) {
+                i++;
+            }
+            if (i == 0) {
```
Would using a switch statement be cleaner here? Also, perhaps checking if the command equate to the exact string command (i.e. `command == "event"`) would make your code more readable?

yizhong187 commented on src/main/java/twilight/AddCommand.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Override
+    public void execute(TaskList tasks, Storage storage) throws InvalidInputException {
+        String addition = "";
```
Is it possible to give the string a more descriptive name?

AbdulrahmanAlRammah replied:
For string that are part of a method I think there is more leeway so I will stick to this.

yizhong187 commented on src/main/java/twilight/AddCommand.java:
```diff
@@ -0,0 +1,36 @@
+            String[] split = details.split(" /by ");
+            addition = tasks.add(new Deadline(split[0], split[1]));
+        }
+        System.out.println(addition);
```
Since you already created the `Ui` abstraction with a `print`, should you use that instead? Or should you abstract out all the UI handling into the `ui` object instantitated in the Twilight` class?

AbdulrahmanAlRammah replied:
Yeah I think I should. I just didn't do it thus far out of laziness.

yizhong187 commented on src/main/java/twilight/Parser.java:
```diff
@@ -0,0 +1,53 @@
+ * Represents a class used for transforming user input to a command to be carried out by chatbot.
+ */
+public class Parser {
+    private static final String[] COMMANDS = new String[]{"list", "mark", "unmark", "todo", "event", "deadline", "delete", "find", "bye"};
```
Perhaps you would want to wrap this line as it is quite long.

yizhong187 commented on src/main/java/twilight/Twilight.java:
```diff
@@ -0,0 +1,29 @@
+    public static void main(String[] args) {
+        Ui ui = new Ui();
+        Scanner input = new Scanner(System.in);
+        Storage storage = new Storage("data/Twilight.txt");
```
Any reason why you did not abstract out the file path as a constant?

yizhong187 commented on src/main/java/twilight/Task.java:
```diff
@@ -0,0 +1,60 @@
+        }
+    }
+
+    public void setDone() {
```
According to the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html): 
> Setter methods for boolean variables must be of the form:
> `void setFound(boolean isFound);`

Perhaps you can lump `setDone` and `setUndone` into one `setDone` method? Or rename the methods to something along the lines of `mark` and `unmark`?

yizhong187 reviewed (commented):
I think you can adhere to the [coding standard ](https://se-education.org/guides/conventions/java/intermediate.html) more strictly. Additionally, perhaps packaging some related files together would improve the project structure?

ulricolo7 commented on src/main/java/twilight/TaskList.java:
```diff
@@ -0,0 +1,117 @@
+     * @return A list of all entries which match the query.
+     */
+    public String query(String q) {
+        TaskList match = new TaskList();
```
Would it be better if the variable is in plural form? i.e. matches

ulricolo7 commented on src/main/java/twilight/Storage.java:
```diff
@@ -0,0 +1,77 @@
+        this.filePath = filePath;
+    }
+
+    /**
```
I might be wrong, but shouldn't there be an empty line after the description and before the @param?

ulricolo7 reviewed (commented):
LGTM! All other comments have been mentioned by the other guy haha! Also, good job consistently adding comments for every method!

AbdulrahmanAlRammah committed:
Add Basic GUI functionality

AbdulrahmanAlRammah committed:
Adjust GUI to work successfully

AbdulrahmanAlRammah committed:
Add greeting and bye to GUI

AbdulrahmanAlRammah committed:
Merge branch 'branch-Level-10'

AbdulrahmanAlRammah committed:
Fix PR review comments

AbdulrahmanAlRammah committed:
Add checkstyle and fix changes

AbdulrahmanAlRammah committed:
Add Assertion

AbdulrahmanAlRammah committed:
Merge pull request #1 from AbdulrahmanAlRammah/branch-A-Assertions

Add Assertion

AbdulrahmanAlRammah committed:
Merge branch 'master' of https://github.com/AbdulrahmanAlRammah/ip

AbdulrahmanAlRammah committed:
Fix code quality issues

AbdulrahmanAlRammah committed:
Merge pull request #2 from AbdulrahmanAlRammah/branch-A-CodeQuality

Fix code quality issues

There are violations to code quality in the lines of code.

The code should be easier to understand and more intuitive.

Apply coding standards taught in CS2103 to improve readability.

AbdulrahmanAlRammah committed:
Set up continuous integration

AbdulrahmanAlRammah committed:
Add extension C-tagging

There is no tagging available.

Tagging is a useful feature that users might want.

Add ability to add multiple tags through the command:
tag tasknum tagdetails. This command can be used multiple times to
add different tags.

AbdulrahmanAlRammah committed:
Merge branch 'branch-BCD-Extension'

AbdulrahmanAlRammah committed:
Address Misinputs

AbdulrahmanAlRammah committed:
Add User Guide

AbdulrahmanAlRammah committed:
Add Help Command

AbdulrahmanAlRammah committed:
Add Ui Png

AbdulrahmanAlRammah committed:
Improve User Guide

AbdulrahmanAlRammah committed:
Improve UG

AbdulrahmanAlRammah committed:
Fix UG formatting

AbdulrahmanAlRammah committed:
Add more JUNIT testing

AbdulrahmanAlRammah committed:
Improve UG

AbdulrahmanAlRammah committed:
Change storage filepath
